manifests/plugin/remote_execution/ssh_user.pp

```diff
@@ -0,0 +1,78 @@
+# This class replicates the functionality of the remote_execution_ssh_keys community templates snippet
+#
+# === Parameters:
+#
+# $manage_user::           Whether to manage the ssh user.  Defaults to the host's Foreman ENC parameter `remote_execution_create_user`.
+#
+# $ssh_user::              The ssh user name. Defaults to the host's Foreman ENC parameter `remote_execution_ssh_user`.
+#
+# $effective_user_method:: Method to switch from ssh user to effective user.  Defaults to the host's Foreman ENC parameter `remote_execution_effective_user_method`.
+#
+# $ssh_keys::              An array of public keys to put in ~/.ssh/authorized_keys.  Defaults to the host's Foreman ENC parameter `remote_execution_ssh_keys`.
+#
+class foreman_proxy::plugin::remote_execution::ssh_user (
+  Boolean            $manage_user           = $foreman_proxy::plugin::remote_execution::ssh_user::params::manage_user,
+  String[1]          $ssh_user              = $foreman_proxy::plugin::remote_execution::ssh_user::params::ssh_user,
+  Enum['sudo', 'su'] $effective_user_method = $foreman_proxy::plugin::remote_execution::ssh_user::params::effective_user_method,
+  Array[String[1]]   $ssh_keys              = $foreman_proxy::plugin::remote_execution::ssh_user::params::ssh_keys,
+) inherits foreman_proxy::plugin::remote_execution::ssh_user::params {
+
+  # Manage the user
+  if $manage_user and $ssh_user != 'root'{
+    user { $ssh_user:
+      ensure           => present,
+      home             => "/home/${ssh_user}",
+      managehome       => true,
+      expiry           => absent,
+      password_max_age => -1, # password never expires
+      password         => '!!',
+      purge_ssh_keys   => true,
+    }
+  }
+
+  # Manage the user's keys
+  $ssh_keys.each |String $ssh_key| {
+    $key_array = split($ssh_key, ' ')
+    $key_type  = $key_array[0]
+    $key       = $key_array[1]
+    $key_name  = $key_array[2]
+
+    ssh_authorized_key { $key_name:
+      ensure => present,
+      user   => $ssh_user,
+      type   => $key_type,
+      key    => $key,
+    }
+  }
+
+  # Manage the sudo configuration
+  if $ssh_user != 'root' and $effective_user_method == 'sudo' {
+
+    $content = "${ssh_user} ALL = (root) NOPASSWD : ALL\nDefaults:${ssh_user} !requiretty\n"
+
+    # saz/sudo is a very popular module
+    # If it's in the environment, go ahead and use it
+    if extlib::has_module('saz/sudo') {
+      sudo::conf { $ssh_user:
+        content => $content,
+      }
+      # Clean up file created by the provisioning template
+      # Most of the time this is probably not necessary as saz/sudo defaults to purging unmanaged files in /etc/sudoers.d
+      file { "/etc/sudoers.d/${ssh_user}":
+        ensure => absent,
+      }
+    } else {
+      if $facts['os']['family'] =~ /(Redhat|Debian)/ {
```

Comment:
We have some code to manage sudoers already and that works for all distros:

https://github.com/theforeman/puppet-foreman_proxy/blob/f10af751d411a45e03537dc1832337eb775366d9/manifests/config.pp#L130-L147

Reply:
I don't think I can use those templates and configuration variables though.  This class is standalone to be included on hosts that aren't otherwise including `foreman_proxy`.
Maybe I should just remove the ` $facts['os']['family'] =~ /(Redhat|Debian)/` conditional?  It was based on the conditional in the [provisioning template](https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/snippet/remote_execution_ssh_keys.erb#L60-L66), but a quick google suggests Suse can also use sudoers.d.

Maybe someone can contribute some augeas for a `/etc/sudoers` implementation later?

Reply:
I'm also fine with hard failures if there's no augeas. You could include `foreman_proxy::params` in this file to determine `$foreman_proxy::params::sudoers` as a base for `sudoers.d`? That way you'd get BSD support as well. I think all linux distros default to having a `sudoers.d` so that's a good enough start for me.

Reply:
Like https://github.com/theforeman/puppet-foreman_proxy/pull/483/commits/a0f24cc2d767eabdc4416e36e6d6f9e6567b55b2 ?

It that's what you were thinking and looks good, I'll rebase/squash.